README.md

```diff
@@ -1,20 +1,105 @@
 # pass
 
-This is a prototype for a password delivery system from a service provider to any users that need secrets.
+This is a text delivery system that someone can utilize to securely send shared
+secrets or password or even just arbitrary text to someone else.
 
-This system has the downside of having credentials at rest in a recoverable way, in the database.
+The system tries to minimize the attack surface on the secrets by encrypting
+data at rest with a token that isn't stored on the system, and the token is only
+provided to the system by a user requesting the data associated with it, and
+when retrieving the data it is deleted as part of the same database query.
 
-## TODO
+## Goals
 
-* Password "masking" in the database
-  * Hashing is not possible due to needing plaintext
-  * Encrypting is possible but the server needs the plaintext so maybe it's useless
-  * Maybe a simple way is just using base64?
-* Actual CSS
-* If the frontend is exposed to public, some rate limit methods should be investigated
+Maintain project simplicity while also securely handling secrets with reversible
+encryption.
 
-## Goals
+## How?
+
+When an entry is created, it gets a Version 4 UUID generated as a unique key
+to find it later.
+
+The UUID is then used as a key for PostgreSQLs `pgp_sym_encrypt`* function to
+encrypt the actual secret data.
+
+But we don't want to store the decryption key right next to the data, so to
+mask it from the database, we pass the UUID, along with a site wide secret
+(salt?), to PostgreSQLs `hmac`* function and store the result as our lookup key.
+
+By doing things this way, we don't have the decryption key for any entry unless
+a request from a user provides it to us, we then pass it to `hmac` again to find
+an entry that matches the same key hash, and finally using the UUID as a key, we
+decrypt the payload field and return it to the requesting client.
+
+## Security
+
+The best case scenario to transmit a secret to someone is to have already
```

Comment:
Start with "In an ideal world," instead.

Reply:
Fixed in next push